### ISSUE TITLE
Update API/CLI to include Nodes Endpoints

### ISSUE THREAD

**Issue by jaredbroad** [QuantConnect/Lean#4524]
#### Expected Behavior
API included the ability to add and remove nodes.
https://github.com/QuantConnect/Lean/blob/master/Api/Api.cs
https://github.com/QuantConnect/Lean/blob/master/Tests/API/ApiTests.cs

CLI Application Includes Node Expansion:
https://github.com/QuantConnect/quantconnect-cli/tree/master/src/commands

#### Actual Behavior
No Node API CRUD support.

#### Potential Solution
Update API for Nodes. 

```
Endpoint api/v2/nodes/create
post:{
	name: 'new node name',
	organizationId : 'organization hash',
	sku: 'B8-1'  -> internally know as node name
	projectId?: 1233 -> Will automatically add the new node to the project (optional)
}

Endpoint api/v2/nodes/read
post:{
	organizationId : 'organization hash',
}

Endpoint api/v2/nodes/stop
post:{
	nodeId: 'node hash',
	organizationId : 'organization hash',
}

Endpoint api/v2/nodes/delete
post:{
	nodeId: 'node hash',
	organizationId : 'organization hash',
}

Endpoint api/v2/nodes/update
post:{
	nodeId : 'node hash',
	name: 'new node name',
	organizationId : 'organization hash',
}

response: {success : true/false, message: ""}
```

#### Checklist 
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues) 
- [x] I have provided detailed steps to reproduce the issue

**Comment by jmerle** [comment]
I can handle the CLI commands, but I got two blocking questions:
- What does `organizationId` refer to? I know projects have an `organizationId`, but as far as I know users normally never see that id and there is no endpoint to retrieve the organizations a user is in.
- The endpoints listed here don't seem to be working? I grabbed the `organizationId` of one of my projects and ran `qcli api:post --body '{ "organizationId": "<id>" }' nodes/read`, but that simply returned the error page you get when you browse to [https://www.quantconnect.com/api/v2/nodes/read](https://www.quantconnect.com/api/v2/nodes/read) with your browser.

**Comment by jaredbroad** [comment]
@jmerle thank you Jasper we'll take you up on that -- let us get the prototype and tests done to debug all the edge cases and then will be ready for CLI. 

An organization is a new concept in QuantConnect to separate out projects and resources and allow teams to collaborate better on the platform. We'll launch it later this week.

**Comment by jmerle** [comment]
I have added the `nodes:list`, `nodes:stop`, `nodes:update` and `nodes:delete` commands to the CLI. I also attempted to create a `nodes:create`, but the `nodes/create` endpoint always responds with an HTTP 500 error when seemingly valid input is given (it responds with descriptive error messages when an invalid SKU is given though).

In my opinion it should be fine without a `nodes:create` command for now, so all actions that can increase an organization's billing are performed in the web interface where the billing change is better visible than in a CLI.
